src/sinks/aws_s3/config.rs

```diff
@@ -141,13 +148,28 @@ impl S3SinkConfig {
             .filename_append_uuid
             .unwrap_or(DEFAULT_FILENAME_APPEND_UUID);
 
+        let transformer = self.encoding.transformer();
+        let (framer, serializer) = self.encoding.clone().encoding();
+        let framer = match (framer, &serializer) {
+            (Some(framer), _) => framer,
+            (None, Serializer::Json(_)) => CharacterDelimitedEncoder::new(b',').into(),
+            (None, Serializer::Native(_)) => {
+                // TODO: We probably want to use something like octet framing here.
+                return Err("Native encoding is not implemented for this sink yet".into());
+            }
+            (None, Serializer::NativeJson(_) | Serializer::RawMessage(_)) => {
+                NewlineDelimitedEncoder::new().into()
+            }
```

Comment:
@lukesteensen same question here as in https://github.com/vectordotdev/vector/pull/12179#discussion_r859610960

Reply:
Same answer as in https://github.com/vectordotdev/vector/pull/12179#discussion_r859874734 😄

Reply:
Same 🙂 

https://github.com/vectordotdev/vector/pull/12179#discussion_r859929618